zerver/models.py

```diff
@@ -2144,6 +2144,18 @@ def get_user_by_delivery_email(email: str, realm: Realm) -> UserProfile:
     return UserProfile.objects.select_related().get(
         delivery_email__iexact=email.strip(), realm=realm)
 
+def get_users_by_delivery_email(emails: Set[str], realm: Realm) -> QuerySet:
+    """This is similar to get_users_by_delivery_email, and
+    it has the same security caveats.  It gets multiple
+    users and returns a QuerySet, since most callers
+    only will grab two or three fields.
+    """
```

Comment:
Let's also note that it doesn't do a `select_related`, so be careful to not access foreign keys like Realm after fetching it.

Reply:
I added some wording here, but you should probably edit it.  I think the main point to emphasize is that "most callers only will grab two or three fields", and we maybe just need to say "should only grab two..." instead of "will only grab two".